pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -37,131 +39,179 @@ func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
-					t.Errorf("pod was recorded incorrectly. expect: %#v, got: %#v", expected, actual)
-				}
-				found = true
-				break
-			}
 		}
-		if !found {
-			t.Errorf("pod %q was not found in %#v", expected.UID, actualPods)
+		normalPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("normal-pod-uid"),
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
 		}
+	)
+
+	testCase := []struct {
+		name         string
+		podList      []*v1.Pod
+		wantPod      *v1.Pod
+		expectPods   []*v1.Pod
+		expectGetPod *v1.Pod
+		expectUID    types.UID
+	}{
+		{
+			name:         "Get normal pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      normalPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: normalPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
+		{
+			name:         "Get static pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      staticPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: staticPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
 	}
-	// Tests UID translation works as expected. Convert static pod UID for comparison only.
-	if uid := podManager.TranslatePodUID(mirrorPod.UID); uid != kubetypes.ResolvedPodUID(staticPod.UID) {
-		t.Errorf("unable to translate UID %q to the static POD's UID %q; %#v",
-			mirrorPod.UID, staticPod.UID, podManager.mirrorPodByUID)
-	}
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+			sort.Slice(actualPods, func(i, j int) bool {
+				return actualPods[i].Name < actualPods[j].Name
+			})
+			if !reflect.DeepEqual(actualPods, test.expectPods) {
+				t.Errorf("Test case %s failed: actualPods and test.expectPods are not equal", test.name)
+			}
 
-	// Test the basic Get methods.
-	actualPod, ok := podManager.GetPodByFullName("bar_default")
-	if !ok || !reflect.DeepEqual(actualPod, staticPod) {
-		t.Errorf("unable to get pod by full name; expected: %#v, got: %#v", staticPod, actualPod)
-	}
-	actualPod, ok = podManager.GetPodByName("default", "bar")
-	if !ok || !reflect.DeepEqual(actualPod, staticPod) {
-		t.Errorf("unable to get pod by name; expected: %#v, got: %#v", staticPod, actualPod)
+			// Tests UID translation works as expected. Convert static pod UID for comparison only.
+			if uid := podManager.TranslatePodUID(mirrorPod.UID); uid != kubetypes.ResolvedPodUID(test.expectUID) {
+				t.Errorf("unable to translate UID %q to the static POD's UID %q; %#v",
+					mirrorPod.UID, staticPod.UID, podManager.mirrorPodByUID)
+			}
+			fullName := fmt.Sprintf("%s_%s", test.wantPod.Name, test.wantPod.Namespace)
+			// Test the basic Get methods.
+			actualPod, ok := podManager.GetPodByFullName(fullName)
+			if !ok || !reflect.DeepEqual(actualPod, test.expectGetPod) {
```

Comment:
The same as above, and please modify all the following `reflect.DeepEqual`.

Reply:
fixed